airbyte-integrations/connectors/destination-shelby/src/main/kotlin/HttpPipelineStep.kt

```diff
@@ -0,0 +1,193 @@
+    override fun getOutputKey(inputKey: StreamKey, output: HttpStepOutput<DestinationRecordRaw>):
+        StreamKey = StreamKey(stream = inputKey.stream)
+
+    override fun getPart(outputKey: StreamKey, inputPart: Int, numParts: Int): Int {
+        return inputPart
+    }
+}
+
+data class HttpStepOutput<T>(
+    override val state: BatchState,
+    val failedRecords: List<T> = emptyList(),
+) : WithBatchState
+
+class HttpPipelineStep(
+    override val numWorkers: Int,
+    private val outputQueue: PartitionedQueue<PipelineEvent<StreamKey, DestinationRecordRaw>>,
+    private val taskFactory: LoadPipelineStepTaskFactory,
+) : LoadPipelineStep {
+    override fun taskForPartition(partition: Int): Task {
+        return taskFactory.createFirstStep(
+            batchAccumulator = HttpAccumulator(),
+            outputPartitioner = NoopPartitioner(),
+            outputQueue = FlattenQueueAdapter(outputQueue),
+            part = partition,
+            numWorkers = numWorkers,
+        )
+    }
+}
+
+data class HttpRequestState(
+    val records: MutableList<DestinationRecordRaw> = mutableListOf(),
+    val outputStream: ByteArrayOutputStream,
+    val printer: CSVPrinter,
+    override val state: BatchState,
+) : WithBatchState, AutoCloseable {
+    companion object {
+        fun new(): HttpRequestState {
+            val outputStream = ByteArrayOutputStream()
+            return HttpRequestState(
+                outputStream = outputStream,
+                printer = CSVPrinter(PrintWriter(outputStream, true, StandardCharsets.UTF_8), CSVFormat.DEFAULT.withQuoteMode(QuoteMode.NON_NUMERIC)),
+                state = BatchState.PROCESSED,
+            )
+        }
+    }
+
+    override fun close() {
+        KotlinLogging.logger {  }.warn { "close" }
+    }
+}
+
+class HttpAccumulator :
+BatchAccumulator<
+    HttpRequestState,
+    StreamKey,
+    DestinationRecordRaw,
+    HttpStepOutput<DestinationRecordRaw>,
+    >
+{
+    override suspend fun start(key: StreamKey, part: Int): HttpRequestState {
+        KotlinLogging.logger {  }.warn { "start $key $part" }
+        return HttpRequestState.new()
+    }
+
+    override suspend fun accept(
+        input: DestinationRecordRaw,
+        state: HttpRequestState,
+    ): BatchAccumulatorResult<HttpRequestState, HttpStepOutput<DestinationRecordRaw>> {
+        val schema = (input.schema as? ObjectType) ?: throw IllegalArgumentException("schema isn't on ObjectType")
+        state.records.add(input)
+        state.printer.printRecord((input.asDestinationRecordAirbyteValue().data as ObjectValue).toCsvRecord(schema))
+        state.printer.flush()
+        if (state.outputStream.size() > 100) {
+            sendData(state)
+//            // TODO check if upload was successful
+//            return FinalOutput(HttpStepOutput(BatchState.COMPLETE))
+            return IntermediateOutput(HttpRequestState.new(), HttpStepOutput(BatchState.LOADED, state.records))
```

Comment:
Another question on this: What is the format of `state.records` needed? If we want to add related HTTP request and responses later, how will this look like?